controller/appcontroller.go

```diff
@@ -2120,7 +2212,7 @@ func alreadyAttemptedSync(app *appv1.Application, commitSHA string, commitSHAsMS
 	} else {
 		// Ignore differences in target revision, since we already just verified commitSHAs are equal,
 		// and we do not want to trigger auto-sync due to things like HEAD != master
-		specSource := app.Spec.Source.DeepCopy()
+		specSource := app.Spec.GetSource()
```

Comment:
DeepCopy was removed.
```suggestion
		specSource := app.Spec.GetSource().DeepCopy()
```

Reply:
I think I did that on purpose, but I can't remember why.

Either way, this code is weird. Why are we emptying fields on temporary vars that we promptly discard?